.circleci/config.yml

```diff
@@ -132,15 +132,19 @@ jobs:
       - run:
           name: set up environment
           command: |
+            python3 -m venv venv
+            . venv/bin/activate
             pip3 install --upgrade setuptools wheel
             pip3 install -r requirements_dev.txt -c constraints.txt
       - save_cache:
           key: v2-pip_cache-{{ checksum "requirements_dev.txt" }}-{{ checksum "constraints.txt" }}
           paths:
             - /home/circleci/.cache/pip
+            - venv
       - run:
           name: run tests
           command: |
+            . venv/bin/activate
```

Comment:
Perhaps use `source` instead of `.`, since `.` is a bash-ism (only bash and maybe zsh support it) so is not POSIX compliant.